pytorch_lightning/utilities/reproducibility.py

```diff
@@ -0,0 +1,21 @@
+"""Helper functions to help with reproducibility of models """
+import os
+
+import numpy as np
+import random
+import torch
+
+
+def seed_everything(seed: int = None):
+    """Function that sets seed for pseudo-random number generators  """
+    try:
+        seed = int(seed)
+    except (TypeError, ValueError):
+        return
+    random.seed(seed)
+    os.environ["PYTHONHASHSEED"] = str(seed)
+    np.random.seed(seed)
```

Comment:
Integers in Python are somewhat flexible. But for NumPy seeds must be convertible to uint32. Maybe we should add checks for each of the framework when needed?

Reply:
Added two assertions for that